clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefMemberChecker.cpp

```diff
@@ -174,7 +175,15 @@ class RawPtrRefMemberChecker
     if (!PropType)
       return;
 
-    auto IsUnsafePtr = isUnsafePtr(QT);
+    if (const ObjCInterfaceDecl *ID = dyn_cast<ObjCInterfaceDecl>(CD)) {
+      if (!RTC || !RTC->defaultSynthProperties() ||
+          ID->isObjCRequiresPropertyDefs())
+        return;
+    }
+
+    bool ignoreARC =
```

Comment:
As a reader unfamiliar with the checker, I'd appreciate some comments on `ignoreARC`.  Specifically, 1) why to `ignoreARC` for non-read-only and `assign` `PD`s; and 2) does `ignoreARC` make it "more likely" an unsafe ptr?